src/server/svr_migrate_data.c

```diff
@@ -771,9 +771,12 @@ rm_migrated_files(char *dirname)
 				if (strcmp(psuffix, JOB_SCRIPT_SUFFIX) == 0) {
 					/* this is a script file, check if .bd file exists */
 					baselen = strlen(path) - strlen(JOB_SCRIPT_SUFFIX);
-					strncpy(pathbd, path, baselen);
-					pathbd[baselen]=0; /* put null char */
-					strcat(pathbd, JOB_BAD_SUFFIX);
+					strncpy(pathbd, path, sizeof(pathbd) - 1);
+					if (baselen < (sizeof(pathbd) - 1))
```

Comment:
Might want to save the value of 'sizeof(pathbd) - 1' since you repeat it twice.

Reply:
Done.